fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -2293,8 +2332,91 @@ struct ShardedRocksDBKeyValueStore : IKeyValueStore {
 					}
 				}
 			} else if (format == RocksDBKeyValues) {
-				a.done.sendError(not_implemented());
-				return;
+				// Make sure the files are complete for the desired ranges.
+				std::vector<KeyRange> fetchedRanges;
+				std::vector<KeyRange> intendedRanges(a.ranges.begin(), a.ranges.end());
+				std::vector<RocksDBCheckpointKeyValues> rkvs;
+				for (const auto& checkpoint : a.checkpoints) {
+					rkvs.push_back(getRocksKeyValuesCheckpoint(checkpoint));
+					for (const auto& file : rkvs.back().fetchedFiles) {
+						fetchedRanges.push_back(file.range);
+					}
+				}
+				std::sort(fetchedRanges.begin(), fetchedRanges.end(), KeyRangeRef::ArbitraryOrder());
+				std::sort(intendedRanges.begin(), intendedRanges.end(), KeyRangeRef::ArbitraryOrder());
+				int i = 0, j = 0;
+				while (i < fetchedRanges.size() && j < intendedRanges.size()) {
```

Comment:
Can you add a comment for this `while` block to explain the logic here?

Reply:
+1 Could you also add comment to explain why `fetchedRanges ` and `intendedRanges` could be different?

Reply:
Added comments, with this kv format, `fetchedRanges` and `intendedRanges` should be the same.